File: api/consumables.py
"""Vercel Serverless Function — /api/consumables/*"""

from __future__ import annotations

from api._lib import logic
from api._lib.base_handler import APIHandler
from api._lib.logic import BadRequest, NotFound


class handler(APIHandler):
    def route(self, method, path, query):
        # /api/consumables
        if path == "/api/consumables":
            if method == "GET":
                return logic.list_consumables()
            if method == "POST":
                return logic.create_consumable(self._read_json())

        # /api/consumables/alerts/low-stock  (고정 경로 우선 매칭)
        if path == "/api/consumables/alerts/low-stock" and method == "GET":
            return logic.low_stock_alerts()

        # /api/consumables/{cid}
        prefix = "/api/consumables/"
        if path.startswith(prefix):
            tail = path[len(prefix):]
            if "/" in tail:
                raise NotFound(f"no route: {method} {path}")
            try:
                cid = int(tail)
            except ValueError:
                raise BadRequest("invalid id")
            if method == "GET":
                return logic.get_consumable(cid)
            if method == "PATCH":
                return logic.update_consumable(cid, self._read_json())
            if method == "DELETE":
                return logic.delete_consumable(cid)

        raise NotFound(f"no route: {method} {path}")

File: api/consumables.js
// Vercel Serverless Function — /api/consumables/*
import * as logic from './_lib/logic.js';
import { BadRequest, NotFound } from './_lib/logic.js';
import { parseUrl, readBody, runHandler } from './_lib/respond.js';
import { authenticateRequest } from './_lib/auth.js';

export default async function handler(req, res) {
  await runHandler(req, res, async () => {
    const { path } = parseUrl(req);
    const method = req.method;

    const { user, householdId } = await authenticateRequest(req);
    const userId = user?.id || null;

    // /api/consumables
    if (path === '/api/consumables') {
      if (method === 'GET') return await logic.listConsumables(householdId, userId);
      if (method === 'POST') {
        const body = readBody(req);
        body.household_id = householdId;
        return await logic.createConsumable(body, userId);
      }
    }

    // /api/consumables/alerts/low-stock
    if (path === '/api/consumables/alerts/low-stock' && method === 'GET') {
      return await logic.lowStockAlerts(householdId, userId);
    }

    // /api/consumables/{cid}
    const prefix = '/api/consumables/';
    if (path.startsWith(prefix)) {
      const tail = path.slice(prefix.length);
      if (tail.includes('/')) throw new NotFound(`no route: ${method} ${path}`);
      const cid = parseInt(tail, 10);
      if (Number.isNaN(cid)) throw new BadRequest('invalid id');
      if (method === 'GET')    return await logic.getConsumable(cid);
      if (method === 'PATCH')  return await logic.updateConsumable(cid, readBody(req), userId);
      if (method === 'DELETE') return await logic.deleteConsumable(cid);
    }

    throw new NotFound(`no route: ${method} ${path}`);
  });
}
